File: internal/messages/message.go
package messages

import (
	"time"

	"github.com/google/uuid"
)

type Message struct {
	ID        uuid.UUID
	UserID    uuid.UUID
	ChatID    uuid.UUID
	Content   string
	CreatedAt time.Time
	Files     []MessageFile
	ReadAt    *time.Time
}

type MessageFile struct {
	ID        uuid.UUID `json:"id"`
	FileKey   string    `json:"file_key"`
	FileName  string    `json:"file_name"`
	MimeType  string    `json:"mime_type"`
	SizeBytes int64     `json:"size_bytes"`
}

File: web/src/lib/models/message.ts
export type Message = {
    id: string;
    fromMe: boolean;
    text: string;
    createdAt: Date;
    readAt: Date | null;
    files?: MessageFile[];
};

export type MessageFile = {
    id: string;
    name: string;
    mimeType: string;
    sizeBytes: number;
};